src/sg_client.rs

```diff
@@ -105,6 +130,57 @@ impl SGClient {
             Ok(res.text()?)
         }
     }
+
+    /// Sends a messages through the SendGrid API. It takes a Mail struct as an argument. It returns
+    /// the string response from the API as JSON.
+    /// 
+    /// ### Example
+    /// ```rust
+    /// # #[tokio::main]
+    /// # async fn main() {
+    /// use sendgrid::{Mail, SGClient};
+    ///
+    /// let mail = Mail::new()
+    ///     .add_from("my-email@address.com")
+    ///     .add_text("hi!")
+    ///     .add_to(("your-email@address.com", "Your Name").into());
+    /// let response = SGClient::new("MY_SECRET_KEY")
+    ///     .send(mail)
+    ///     .await
+    ///     .unwrap();
+    /// # }
+    /// ```
+    #[cfg(feature = "async")]
+    pub async fn send(&self, mail_info: Mail<'_>) -> SendgridResult<String> {
+        let mut headers = HeaderMap::new();
```

Comment:
We may want to make this header construction into its own function if it is common across synchronous and asynchronous versions.

Reply:
Done!